src/main/java/de/tum/cit/aet/artemis/assessment/repository/cleanup/FeedbackCleanupRepository.java

```diff
@@ -89,6 +118,36 @@ SELECT MAX(r2.id)
             """)
     int deleteOldFeedbackThatAreNotLatestRatedResultsWhereCourseDateBetween(@Param("deleteFrom") ZonedDateTime deleteFrom, @Param("deleteTo") ZonedDateTime deleteTo);
 
+    /**
+     * Counts {@link Feedback} entries associated with rated {@link Result} that are not the latest rated result
+     * for a {@link Participation}, within courses conducted between the specified date range.
+     *
+     * @param deleteFrom the start date for selecting courses
+     * @param deleteTo   the end date for selecting courses
+     * @return the number of entities that would be deleted
+     */
+    @Query("""
+            SELECT COUNT(f)
+            FROM Feedback f
+            WHERE f.result IN (
```

Comment:
same here, might become quite costly

Reply:
yes agree, they are quite expensive but cleanup operations will probably be executed 1-2 a year, I don't think that's a big performance issue